### PR TITLE
7-multi-arch-deployment-to-docker

### PR DESCRIPTION
**Pull request by thomaszub**

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/.devcontainer
+/.github
+/.vscode
```
Comment (aserowy) [inline .dockerignore:3]: this would remove vscode dev environment and pipelines :)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -5,104 +5,76 @@ on:
...
-          override: true
-          profile: minimal
-          toolchain: stable
-
-      - name: security_audit
-        uses: actions-rs/audit-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-
-  check:
```
Comment (aserowy) [inline .github/workflows/release.yml:27]: Why should checks, lint, and tests be removed?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -157,9 +115,23 @@ jobs:
...
+            --build-arg BASE_IMAGE=${{ matrix.base_image }}:${{ matrix.version }} \
+            --build-arg SUB_DIR=${{ matrix.arch_rust }}-unknown-linux-${{ matrix.libc }}/ \
+            .
+
+      - name: docker_build_and_push_latest
+        if: ${{ matrix.tag_as_latest }}
         run: |
           docker buildx build --push \
-            --tag serowy/docker2mqtt:${{ steps.version_tag.outputs.version }} \
             --tag serowy/docker2mqtt:latest \
```
Comment (aserowy) [inline .github/workflows/release.yml:132]: We can push multiple tags with one build.

File: .dockerignore (hunk 1 above)
Comment (thomaszub) [reply .dockerignore:3]: Okay, will fix this

File: .github/workflows/release.yml (hunk 2 above)
Comment (thomaszub) [reply .github/workflows/release.yml:27]: DRY, every commit in main runs through the ci pipeline, so this is redundant.

File: .github/workflows/release.yml (hunk 3 above)
Comment (thomaszub) [reply .github/workflows/release.yml:132]: We have to separate it as we can tag Debian and Alpine both as latest.

Comment (aserowy) [reply .github/workflows/release.yml:132]: Latest-alpine
Latest-...

Should be sufficient?

File: .github/workflows/release.yml (hunk 2 above)
Comment (aserowy) [reply .github/workflows/release.yml:27]: Didn't thought about it.

File: .github/workflows/release.yml (hunk 3 above)
Comment (thomaszub) [reply .github/workflows/release.yml:132]: Hmm, would be possible, but latest without suffix is a Docker standard for the default image. Would not drop it for convention reasons.

**Review by aserowy — APPROVED** [review]

**Pull request merged**